main.tf

```diff
@@ -3,40 +3,9 @@ resource "azurerm_firewall_policy_rule_collection_group" "this" {
   firewall_policy_id = var.firewall_policy_id
   priority           = 100
 
-  network_rule_collection {
-    name     = "rc-DomainController-${var.stage}"
-    priority = 100
-    action   = "Allow"
-
-    rule {
-      name                  = "allow-alz-to-dc-inbound"
-      protocols             = ["TCP", "UDP"]
-      source_ip_groups      = [var.ipg_application_lz_id]
-      destination_ip_groups = var.ipg_onpremise_dc_id != null ? [var.ipg_azure_dc_id, var.ipg_onpremise_dc_id] : [var.ipg_azure_dc_id]
-      destination_ports = [
-        "53", "88", "123", "135", "137", "138", "139",
-        "389", "445", "464", "636", "3268", "3269", "9389"
-      ]
-    }
-  }
-
-  network_rule_collection {
-    name     = "rc-DNSPrivateResolver-${var.stage}"
-    priority = 110
-    action   = "Allow"
-
-    rule {
-      name                  = "allow-dc-to-dnsresolver-inbound"
-      protocols             = ["Any"]
-      source_ip_groups      = var.ipg_onpremise_dc_id != null ? [var.ipg_azure_dc_id, var.ipg_onpremise_dc_id] : [var.ipg_azure_dc_id]
-      destination_ip_groups = [var.ipg_dnsprivateresolver_id]
-      destination_ports     = ["*"]
-    }
-  }
-
   network_rule_collection {
     name     = "rc-internet_outbound-${var.stage}"
-    priority = 120
```

Comment:
Priority 100 is reserved by platform. So no collisions there.

Rules were re-ordered because all network rules need to be in front of application rules. Previously we didn't leave any free space so it was not possible to insert a new rule if we wanted to stick to a 10 diff.

We did a 5 diff to allow for more RCs in total and left space between the end of network rules and the start of application rules.

This change causes no downtime.